source/extensions/filters/common/ext_authz/ext_authz.h

```diff
@@ -58,6 +58,9 @@ struct Response {
   // A set of HTTP headers returned by the authorization server, will be optionally added
   // (using "addCopy") to the request to the upstream server.
   Http::HeaderVector headers_to_add;
+  // A set of HTTP headers consumed by the authorization server, will be removed
+  // (using "remove") from the request to the upstream server.
+  std::vector<Envoy::Http::LowerCaseString> headers_to_remove;
```

Comment:
Can we use `Http::HeaderVector` here?

Reply:
HeaderVector is a `std::vector<std::pair<LowerCaseString, std::string>>`. This is perfect for `headers_to_append`, `headers_to_set` and `headers_to_add`, because those need both a key and a value. `headers_to_remove` only needs the header name (ie. the key). The value would always be empty. So I figured a plain `std::vector<Envoy::Http::LowerCaseString>` would be a better fit.

Happy to change this if you prefer `HeaderVector`?

Reply:
Got it. I think we can keep this as is.